### PR TITLE
storage: add tests for unsafeAddWatching

### PR DESCRIPTION
**Pull request by gyuho**
This adds map operation tests for unsafeAddWatching, which
could have been failed without https://github.com/coreos/etcd/pull/3939.

/cc @xiang90 

Thanks!

### CODE REVIEW

File: storage/watchable_store_test.go [hunk 1]
```diff
@@ -94,4 +94,24 @@ func TestUnsafeAddWatching(t *testing.T) {
 		// when putting the same watch twice"
 		t.Error(`error = nil, want "put the same watch twice"`)
 	}
+
+	// test the map operation
+	ws := &watching{
```
Comment (xiang90) [inline storage/watchable_store_test.go]: can we change this to a simple for loop with the previous one? like add three watching on the same key?

**Comment by gyuho** [comment]
@xiang90 Good point. Just updates tests to use `for` loop. Thanks,

File: storage/watchable_store_test.go [hunk 2]
```diff
@@ -69,29 +69,29 @@ func TestUnsafeAddWatching(t *testing.T) {
...
+			prefix: true,
+			cur:    0,
 		}
 	}
-
-	if err := unsafeAddWatching(&s.synced, string(testKey), wa); err == nil {
-		// unsafeAddWatching should have returned error
-		// when putting the same watch twice"
-		t.Error(`error = nil, want "put the same watch twice"`)
+	// test the map operation
```
Comment (xiang90) [inline storage/watchable_store_test.go]: the comment is not accurate. we are testing the add operations and verifying the result of the operation by checking the map.

**Comment by xiang90** [comment]
Please make the comment more accurate then the code looks good to me. Thanks!

**Comment by gyuho** [comment]
@xiang90 Just updated comments and commit message. Thanks

**Comment by xiang90** [comment]
LGTM

**Pull request merged**